nucliadb/src/nucliadb/search/search/shards.py

```diff
@@ -61,11 +61,41 @@ async def get_shard(node: AbstractIndexNode, shard_id: str) -> Shard:
 async def query_paragraph_shard(
     node: AbstractIndexNode, shard: str, query: ParagraphSearchRequest
 ) -> ParagraphSearchResponse:
-    req = ParagraphSearchRequest()
-    req.CopyFrom(query)
-    req.id = shard
+    req = SearchRequest()
```

Comment:
This is called in a single place `resource_search`. Maybe it makes more sense to put this logic (or a subset of it) in `resource_search` and remove this method altogether?

Reply:
Yes, still working on it. The first step was to remove the calls in the innermost point and go bottom to top

Reply:
Done